code/datums/components/attachment_handler.dm

```diff
@@ -316,13 +318,19 @@
 ///Updates the mob sprite of the attachment.
 /datum/component/attachment_handler/proc/apply_custom(datum/source, image/standing)
 	SIGNAL_HANDLER
+	var/obj/item/parent_item = parent
+	if(!ismob(parent_item.loc))
+		return
```

Comment:
Why is this check needed

Reply:
`wearing_mob.regenerate_icons()`
we want to update the clothing overlays if the parent_item is worn by a mob